### ISSUE TITLE
All Performables below IsSilent Performable are not reported

### ISSUE THREAD

**Issue by nbarrett** [serenity-bdd/serenity-core#1525]
Given a task as follows:

```
public class StartInitiallySilently implements Interaction, IsSilent {

    private final List<String> items;

    public StartInitiallySilently(List<String> items) {
        this.items = items;
    }

    public <T extends Actor> void performAs(T actor) {
        actor.attemptsTo(
            Start.withATodoListContaining(items));
    }

    public static Interaction withATodoListContaining(String... items) {
        return instrumented(StartInitiallySilently.class, newArrayList(items));

    }
}

```

This test will produce no output at all:

```
    @Test
    public void reproduces_problem_with_non_reported_tasks() {
        givenThat(james).wasAbleTo(StartInitiallySilently.withATodoListContaining("Walk the dog", "Put out the garbage"));

    }
```

So it seems that all tasks whether they implement `IsSilent` or not will be hidden from reporting if they are called from an `IsSilent Task`. To me this is not desired/expected and the scope of the 'silence' should be within the tasks that implement the `IsSilent` or `CanBeSilent` interface with  `isSilent() = true`

**Comment by nbarrett** [comment]
I've created a branch to reproduce the issue: https://github.com/nbarrett/screenplay-pattern-todomvc/tree/serenity-core-issue-1525 - command to reproduce problem would be `mvn clean verify -Dtags=issue:issue-1523`

**Comment by wakaleo** [comment]
This is intentional - a silent task doesn’t report nested tasks. If you need to do this we probably need to add another interface (say, IsHidden) to cater for this. It would be a generalisation of the way the conditionals are handled.

**Comment by nbarrett** [comment]
Yes, the exact use-case I have which is causing problems on my current client project is this:

```
public class LoginToAgentPortalIfRequired implements Interaction, IsSilent {

    @Override
    public <T extends Actor> void performAs(T actor) {
        TestAgent testAgent = AuthenticateAgent.as(actor).testAgent;
        actor.attemptsTo(
            Check.whether(userIsLoggedIn(testAgent.getUsername()))
                .otherwise(Login.toAgentPortal()));
    }
}
```

The `Login.toAgentPortal()` tasks are never reporting under these circumstances, when that is desired behaviour. If I turn off `IsSilent` then I get undesirable and grammatically incorrect `agent login to agent portal if required` littering my test reports as documented in #1491.

**Comment by wakaleo** [comment]
You can now use the `IsHidden` interface for this.

**Comment by nbarrett** [comment]
Thanks @wakaleo - I saw your commit the other day and plan to test and release this to my client in the next day or so  - much appreciated!

**Comment by nbarrett** [comment]
Sorry for the immensely long feedback cycle on this one! I've just upgraded to Serenity core 2.0.60 and  it seems that `IsHidden` still doesn't work. For example in this `Performable`:

```
public class CreateAccount implements Performable, IsHidden {

    @Step("{0} creates an account")
    @Override
    public <T extends Actor> void performAs(T actor) {
        actor.attemptsTo(
            Close.anyOpenClaimantSessions(),
            Complete.postcodeChecker());
    }
}
```
I see this in the report:

![image](https://user-images.githubusercontent.com/626941/61869523-b6388280-aed3-11e9-88ab-44c80891153a.png)

Note that `Close.anyOpenClaimantSessions()` does not report anything as I've used `IsSilent`, so that marker interface is working. I'll try and create an example project to go along with this.
